### PR TITLE
Link to casa

### PR DESCRIPTION
**Pull request by gijzelaerr**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,8 @@ add_library(stationresponse SHARED
 add_library(elementresponse SHARED
   ElementResponse.cc)
 
+target_link_libraries(stationresponse casa_casa casa_measures casa_tables casa_ms)
```
Comment (aroffringa) [inline CMakeLists.txt:32]: This is not the right way of linking with the casacore libraries. Plus, line 37 already has link statements:
`
target_link_libraries(stationresponse ${CASACORE_LIBRARIES})
target_link_libraries(elementresponse ${CASACORE_LIBRARIES})
`
So this is unnecessary...

**Review by aroffringa — CHANGES_REQUESTED** [review]
There's no need to add this statement. Linking should of course also use the defined macros, and not have hardcoded libraries in them.

**Comment by gijzelaerr** [comment]
aha! I was confused, the `target_link_libraries()` statements are not included in the 4.0 release (of which I'm building a package), so I patched it and did a PR since I assumed they would not be in master also.

Sorry for the noise!

By the way, Debian linting will probably complain about your way of doing this with the macro since `elementresponse` doesn't seem to use any of the symbols on the `casacore` libraries. This means that when the  `libelementresponse` is installed it would automatically also download & install all linked packages, although they are not required. But since it is very likely casascore is installed already anyway this is not really an issue.

**Comment by aroffringa** [comment]
Ok, thanks for the info.

Is the problem solved if you would put the elementresponse lib in the same package as the stationresponse lib? I don't think anyone would want to install one without the other...

I'm happy to remove the elementresponse casacore linking if that's helpful (I'm a bit surprised it doesn't use casacore).

**Comment by gijzelaerr** [comment]
The Debian way of doing things is to make a package per shared library. A lot of work like dependency resolution are automated and work well, as long you stick some simple rules. Hence me always asking astronomers to add the SO numbers. 

About element response, I was surprised too. It seems to work fine without the casacore linking. If you agree you can remove it.

**Pull request closed without merging**